src/Servers/Connections.Abstractions/src/IConcurrentConnectionListener.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Connections;
+
+/// <summary>
+/// Indicates that a <see cref="IConnectionListener"/> or <see cref="IMultiplexedConnectionListener"/> supports concurrent accept operations.
+/// </summary>
+public interface IConcurrentConnectionListener
```

Comment:
Why does this need to be exposed as an interface at all?

Reply:
See James' point 2 above; "because not all listeners might support it" - however, probably not worth worrying about too much yet - I need to `crank` it a bit to see whether this is even useful (figured that out today, and it turns out I didn't even need to commit anything for that, so: this PR is probably way too early and redundant)

Reply:
> This change means concurrent calls to IConnectionListener.AcceptAsync. That's an extension point. It's possible that some user implementations weren't designed for this. I just checked the named pipe transport that I'm working on, and it appears to be ok with concurrent calls. Probably will need to call out as a possible breaking change.


Got it. I think that's a reasonable concern but I'm not sure it warrants an interface just yet. Anyways, once we measure, we'll know more.